packages/@expo/metro-config/src/transform-worker/transform-worker.ts

```diff
@@ -0,0 +1,219 @@
+
+  if (options.dev) {
+    return worker.transform(
+      config,
+      projectRoot,
+      filename,
+      // In development, we use a JS file that appends a style tag to the
+      // document. This is necessary because we need to replace the style tag
+      // when the CSS changes.
+      // NOTE: We may change this to better support static rendering in the future.
+      Buffer.from(wrapDevelopmentCSS({ src: code, filename })),
+      options
+    );
+  }
+
+  const { transform } = await import('lightningcss');
+
+  // TODO: Add bundling to resolve imports
+  // https://lightningcss.dev/bundling.html#bundling-order
+
+  const cssResults = transform({
+    filename,
+    code: Buffer.from(code),
+    sourceMap: false,
+    cssModules: false,
+    projectRoot,
+    minify: options.minify,
+  });
+
+  // TODO: Warnings:
+  // cssResults.warnings.forEach((warning) => {
+  // });
+
+  // Create a mock JS module that exports an empty object,
+  // this ensures Metro dependency graph is correct.
+  const jsModuleResults = await worker.transform(
+    config,
+    projectRoot,
+    filename,
+    Buffer.from(''),
+    options
+  );
+
+  const cssCode = cssResults.code.toString();
+
+  // In production, we export the CSS as a string and use a special type to prevent
+  // it from being included in the JS bundle. We'll extract the CSS like an asset later
+  // and append it to the HTML bundle.
+  const output: JsOutput[] = [
+    {
+      data: {
+        // @ts-expect-error
+        ...jsModuleResults.output[0].data,
+
+        // Append additional css metadata for static extraction.
+        css: {
+          code: cssCode,
+          lineCount: countLines(cssCode),
+          map: [],
+          functionMap: null,
+        },
+      },
+      type: 'js/module',
+    },
+  ];
+
+  return {
+    dependencies: jsModuleResults.dependencies,
+    output,
+  };
+}
+
+/**
+ * A custom Metro transformer that adds support for processing Expo-specific bundler features.
+ * - Global CSS files on web.
+ * - CSS Modules on web.
+ * - TODO: Tailwind CSS on web.
+ */
+module.exports = {
+  transform,
+
+  getCacheKey(config: JsTransformerConfig): string {
+    const { babelTransformerPath, minifierPath, ...remainingConfig } = config;
+
+    const filesKey = getCacheKey([
+      require.resolve(babelTransformerPath),
+      require.resolve(minifierPath),
+      require.resolve('metro-transform-worker/src/utils/getMinifier'),
+      require.resolve('metro-transform-worker/src/utils/assetTransformer'),
+      require.resolve('metro/src/ModuleGraph/worker/generateImportNames'),
+      require.resolve('metro/src/ModuleGraph/worker/JsFileWrapping'),
+      ...metroTransformPlugins.getTransformPluginCacheKeyFiles(),
+    ]);
+
+    const babelTransformer = require(babelTransformerPath);
+    return [
+      filesKey,
+      stableHash(remainingConfig).toString('hex'),
+      babelTransformer.getCacheKey ? babelTransformer.getCacheKey() : '',
+    ].join('$');
```

Comment:
You should replace most of this with a call to the built-in transformer's `getCacheKey` function instead of duplicating its behaviour.